index.js

```diff
@@ -7,37 +7,54 @@ export default class Flatbush {
 
         this.numItems = +numItems;
         this.nodeSize = Math.max(+nodeSize || 16, 2);
-        this.ArrayType = ArrayType || Float64Array;
 
         // calculate the total number of nodes in the R-tree to allocate space for
         // and the index of each tree level (used in search later)
         let n = numItems;
         let numNodes = n;
-        this._levelBounds = [n * 5];
+        this._levelBounds = [n * 4];
         do {
             n = Math.ceil(n / this.nodeSize);
             numNodes += n;
-            this._levelBounds.push(numNodes * 5);
+            this._levelBounds.push(numNodes * 4);
         } while (n !== 1);
 
+        this.ArrayType = ArrayType || Float64Array;
+        this.IndexArrayType = Float64Array;
+        if (numNodes * 4 < Math.pow(2, 8)) {
+            this.IndexArrayType = Uint8Array;
+        } else if (numNodes * 4 < Math.pow(2, 16)) {
+            this.IndexArrayType = Uint16Array;
+        } else if (numNodes * 4 < Math.pow(2, 32)) {
+            this.IndexArrayType = Uint32Array;
+        }
+        const alignmentMargin = Math.max(0, this.IndexArrayType.BYTES_PER_ELEMENT - 4 * this.ArrayType.BYTES_PER_ELEMENT);
```

Comment:
Can you say more about what `alignmentMargin` means and why it's calculated like this? I can't think of a combination of index/box types that would make this value not `0`.

Reply:
I think this is the only case:
* odd number of nodes
* numNodes is very large (index type is `Float64Array`)
* array type is `Int8Array` or `Uint8Array`

This would result in an exception: `RangeError: start offset of Float64Array should be a multiple of 8`, if one doesn't add a small _"margin"_ between the coordinates and the indices.

That said, I see now that my logic was indeed wrong (if the number of nodes is even, the margin should be zero)! Something like this should work: `const alignmentMargin = (this.IndexArrayType.BYTES_PER_ELEMENT - numNodes * 4 * this.ArrayType.BYTES_PER_ELEMENT % this.IndexArrayType.BYTES_PER_ELEMENT) % this.IndexArrayType.BYTES_PER_ELEMENT;` (I'll update the PR in a minute)

Reply:
@tyrasd 👍 let's get rid of it.

Reply:
done